docs/structured-streaming-programming-guide.md

```diff
@@ -861,6 +861,10 @@ isStreaming(df)
 </div>
 </div>
 
+You may want to check the logical plan of the query, as Spark converts the operation into another operation, which includes adding streaming aggregation. (e.g. count, distinct, union, etc.)
```

Comment:
I think the operation converting is internal behavior, maybe it's not clear enough for asking the end-user to check it. How about we just comment on the behavior difference between SQL distinct and dataset dropDuplicate, WDYT?

Reply:
The thing is whether Spark injects streaming aggregation which end users have to maintain or not, and that can be checked by looking into logical plan, right? I didn't mean they need to find the distinct in logical plan and how Spark changes the operation. They just need to check for stateful operations.

SQL distinct and Dataset dropDuplicate aren't the only difference. SQL union and Dataset union are also different. The cases can increase and decrease according to the Spark catalyst rules, which is not we can ensure the doc be in sync.

Reply:
Probably we can reword here as well to simplify, like 

```
You may want to check the query plan of the query, as Spark could inject stateful operations during interpret of the query. Once stateful operations are injected in the query plan, you may need to check your query with considerations in stateful operations. (e.g. output mode, watermark, state store size maintenance, etc.)
```

If the reworded sentences sound better then I can update.

Reply:
The reworded version looks better.

Reply:
Thanks for the input. Will update.